### ISSUE TITLE
Syntax to set a reminder for a task

### ISSUE THREAD

**Issue by idpaterson** [idpaterson/alfred-wunderlist-workflow#66]
Reminders in Wunderlist can be used in a number of ways. They can be added to tasks with due dates to remind the user well in advance of upcoming tasks, they can act as the time component for a task due at a specific time of day, or they can even be used without a due date as a reminder that something needs to get done eventually.

Syntax may be explicit such as a "remind me" keyword or implicit, such as a due date with a time component "due Friday at noon." Currently the time component from parsedatetime is ignored, but the latter could possibly be implemented as simply as handling the times that are already parsed. A reminder may use an absolute date format like May 12, or a relative date format like "2 days in advance." I believe that parsedatetime can handle certain relative time delta expressions but would have to do some experimentation.

If you use reminders and find the omission of this feature to be of grave detriment to your productivity, post a comment below!

**Comment by ckoeberle** [comment]
+1

I miss this feature! Would be great if you implement it.

**Comment by idpaterson** [comment]
How would you prefer to type it in to Alfred?

**Comment by ckoeberle** [comment]
I think the explicit syntax ("remind me", "reminder", "remind", "alarm", ...) makes sense, because you can set a due date different to the reminder's date and time. Should be short enough to type it frequently ;-)

**Comment by idpaterson** [comment]
Do you have any use for a *default* reminder time? For example, you might set the default time to 6:00pm in the workflow preferences then could enter tasks like "Take out the trash every friday remind me" and a reminder would be set for Fridays at 6. Wunderlist seems to have added a default reminder of 9am that is being added for tasks with a due date, so if this feature were useful it would probably default to 9am.

I will probably also add "r" as a keyword if it is the last word in the task followed by just a datetime, for those who don't want to type any more than necessary.

**Comment by ckoeberle** [comment]
personally I don't need a default reminder tim, but it surely is a nice
feature.
I like the "r" keyword :-)

____________________________
*Christoph Köberle*
Scandio GmbH
Charles-de-Gaulle-Str. 4
81737 München
Tel.: +49 (89) 244 124-41
Fax: +49 (89) 244 124-10
Mail: christoph.koeberle@scandio.de
Web: www.scandio.de
HRB 14 88 20 Amtsgericht München

2015-06-17 15:12 GMT+02:00 Ian Paterson <notifications@github.com>:

**Comment by idpaterson** [comment]
Thanks for the input, I was looking for a feature for version 0.5 and I think this is perfect since it rounds out task entry without getting into more complicated things like subtasks and notes. Have been working on 0.6 building blocks lately because it's a bit more exciting (task sync allows searching and completing tasks, showing today's tasks, hashtag suggestion, etc).

**Comment by idpaterson** [comment]
Interesting, it seems that the Wunderlist API allows multiple reminders for each task. I don't see support for that in any of the Wunderlist apps so I don't think I'll worry about supporting it in the workflow yet. Just something to think about, I haven't tested multiple reminders.

**Comment by idpaterson** [comment]
@ckoeberle, please download [this preview build](https://copy.com/kFKp9jkPi8ejL1Yn/Wunderlist-reminders.alfredworkflow?download=1) that includes reminder support. I hope you like it, there is just a bit more to do before posting a public beta release including this feature. Please let me know how this works for you and if you find anything that doesn't work right!

The keyword options are `remind me, reminder, remind, r, alarm` optionally followed by a `:`. Those keywords can be followed by a date or if not followed by a date the default 9AM time is assumed. If the task has a due date, the times are relative to that date otherwise the times are relative to today. The parsed reminder, due date, and recurrence are displayed while you type the task. It should not matter which order you type the reminder, due date, and recurrence phrases.

Some things that don't work yet:

* There is no setting to change the default 9 AM reminder time yet
* The task entry won't allow you to create a task that ends in one of the reminder keywords unless you type quickly enough to not get the reminder prompt (e.g. "pay rent due april 4 remind me" which would set a reminder for 9AM April 4)
* Dates relative to the due date do not work, like "remind me 2 days before" – I'm not sure if I will support that or not

**Comment by ckoeberle** [comment]
wow - great! Thank you very much!
I found one little issue:
``` wl do something due 30.6 r 29.6 7pm ``` results in a reminder at 7am, but
``` wl do something due 30.6 r 29.6 19:00 ``` does the trick

**Comment by idpaterson** [comment]
Interesting, since 30.6 is parsing as a date (I assume June 30, right?) perhaps your workflow is adopting the German locale for parsedatetime. I was not sure whether that worked or not within Alfred but good to know.

The German locale does not support AM/PM (meridian):
```python
        self.meridian      = [ ]
        self.usesMeridian  = False
        self.uses24        = True

**Comment by idpaterson** [comment]
While I have you here, would you be able to provide a few translations that are missing for parsedatetime to improve this workflow and other projects that use it? If so, please let me know what the German translations would be for the following, if the same phrases are used with dates and times:

|  English   |                Meaning                |              Example              |
| ---------- | ------------------------------------- | --------------------------------- |
| `from`     | Follows a reference date              | two days from now                 |
| `after`    | Follows a reference date              | the day after my birthday         |
| `before`   | Precedes a reference date             | one week before my birthday       |
| `previous` | One unit before a reference date      | the previous day                  |
| `prev`     | Abbreviation of `previous`            | the prev day                      |
| `end of`   | The last unit in a specific timeframe | remind me at the end of the month |
| `eod`      | Abbreviation of `end of the day`      | due by eod Friday                 |
| `eom`      | Abbreviation of `end of the month`    | finish by eom                     |
| `eoy`      | Abbreviation of `end of the year`     | remind me eoy                     |
| `a`        | Used in place of "one"                | in a month                        |
| `an`       | Used in place of "one"                | in an hour                        |

Also, the numbers zero through twenty spelled out. If any of the above have common abbreviations you can include those as well.

**Comment by ckoeberle** [comment]
Not too easy, because sometimes the is no single word corresponding to the english word. I'll try ...

English | German | Example (english) | Example (german)
---------- | ---------- | ---------- | ----------
`from` | `in` `ab` | two days from now | `in` zwei Tagen `ab` heute
`after`| `nach` | the day after my birthday | einen Tag `nach`meinem Geburtstag
`before` | `vor` | one week before my birthday | eine Woche `vor` meinem Geburtstag
`prev` | `vorher` | the prev day | einen Tag `vorher`<br>eine Woche `vorher`
`end of` | `Ende des`<br>`Ende der` | remind me at the end of the month<br>remind me at the end of the week | erinnere mich `Ende des` Monats<br>erinnere mich `Ende der` Woche
`eod`<br>`eom`<br>`eoy` | there is no corresponding german abbreviation
`a`<br>`an` | same as `one`<br>`einem`, `einer` | in a month<br>in an hour | in `einem` Monat<br>in `einer` Stunde

#### Numbers 0 to 20

Number | spelled
----- | -----
0 | null
1 | eins
2 | zwei
3 | drei
4 | vier
5 | fünf
6 | sechs
7 | sieben
8 | acht
9 | neun
10 | zehn
11 | elf
12 | zwölf
13 | dreizehn
14 | vierzehn
15 | fünfzehn
16 | sechzehn
17 | siebzehn
18 | achtzehn
19 | neunzehn
20 | zwanzig

**Comment by idpaterson** [comment]
Thanks! I will work on adding these to parsedatetime. I will set up an issue over there to track the changes and might have some follow up questions for you later.

**Comment by vmalloc** [comment]
+1 for the reminders feature. Any idea when it would make its way to the official build? Also, it appears the preview build does not work properly when trying to combine list and reminders: `wl work: do something due today r 13:00`

**Comment by ckoeberle** [comment]
> Also, it appears the preview build does not work properly when trying to combine list and reminders: `wl work: do something due today r 13:00`

Right, all entries go to inbox. I think this is caused by the colon in german time `13:00` (instead of `1pm`)

**Comment by vmalloc** [comment]
sure, but it should take that into consideration (only split the last expression containing a colon for instance)

**Comment by idpaterson** [comment]
@vmalloc, good observation! The list name matching was being too greedy and tried to match `work: do something due today r 13` as the list name. In [this update](https://copy.com/wHNgG38QKYZTndky/Wunderlist-list-and-reminders.alfredworkflow?download=1) the list name will now stop at the first colon.

I am glad to hear that the feature is useful, while I hoped to include some other improvements in 0.5 it is probably best to just get reminders (and the other changes so far in 0.5) out there. A new beta later today (well, *my* today – it's still morning) will include this fix along with a few other minor improvements. I plan to release 0.5 this weekend, just let me know if anything else is broken.

**Comment by ckoeberle** [comment]
Great, that works! 
I found one more, I added a separate issue, because I don't think it has something to do with the reminders.

**Comment by vmalloc** [comment]
Seems to work. Thanks!

**Comment by idpaterson** [comment]
I'm going to wait on that next beta until I add in the new setting discussed in #81
